modules/general_functions.py

```diff
@@ -656,3 +657,51 @@ def find_next(iterable: Iterable[T], cond: Callable[[T], bool]) -> T:
         return next(i for i in iterable if cond(i))
     except StopIteration:
         raise ValueError("Value not found in iterable.")
+
+# FIXME: Find better placement
+class ReferenceDensity:
+    def __init__(self, layers: List = None):
```

Comment:
`layers` must be a mandatory argument, otherwise the both of the methods will throw an error.

Type annotations for sequences are a lot more useful when they include the inner type: `List[Layer]`.

Reply:
OK